database.hh

```diff
@@ -744,7 +744,7 @@ public:
     }
 
     mutation_source as_mutation_source() const;
-    mutation_source as_mutation_source_excluding(sstables::shared_sstable sst) const;
+    mutation_source as_mutation_source_excluding(std::vector<sstables::shared_sstable>& sst) const;
```

Comment:
I'll review this pull request properly on Tuesday after the holidays end here, but I'll leave a small note for now:

I remember that we explicitly wanted to create a specific reader, which reads "all sstables except for this one" in order to generate view updates from an sstable. I don't recall *why* exactly we wanted that though... I'll need to go re-read some design docs and commit messages. The point is, that this commit changes this reader's semantics from "read all sstables except for this one" to "read all sstables except for this bunch of sstables" - and we need to consider if this change of semantics does not cause any problems later, e.g. that we won't read fresh-enough data when generating view updates. 

Considering a corner case of this vector containing every single staging sstable, this new reader will effectively not use any staging sstable when reading data during the view update generation. In the old code, it will still use almost every staging sstable when generating view updates. I'm not sure now *if* it matters and *why* it could matter, so I'll go back to the topic once I refresh my memory about this design.

Reply:
I gave it some thought, and it's all good. It's even better than good, since the original idea was to exclude all sstables coming from the same session, not just the one being processed.

Just to add some fuel to the `list vs github` fire (even though I'm a gerrit fan myself), I feel compelled to add that our mailing list system let me conveniently browse all the 7 versions of the staging sstable patches, and find the explanation of all the decisions in the changelog (:

Reply:
sure, but can you add the polish flag to the review comments in the mailing list?

:poland: